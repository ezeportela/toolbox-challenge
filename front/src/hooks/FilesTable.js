import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { FileRestService } from "../lib/FileRestService";
import { setFilesList, setFilesContent, toggleLoading } from "../store";

export function useFilesTable() {
  const { files, filesList } = useSelector((state) => state);
  const dispatch = useDispatch();

  const fileRestService = new FileRestService();

  async function getFilesList() {
    const response = await fileRestService.getFilesList();
    dispatch(setFilesList(response.data));
  }

  async function getFilesContent() {
    dispatch(toggleLoading());
    const response = await fileRestService.getFilesContent();
    dispatch(setFilesContent(response.data));
    dispatch(toggleLoading());
  }

  useEffect(() => {
    getFilesList();
    getFilesContent();
  }, []);

  const columns = ["File Name", "Text", "Number", "Hex"];

  return { files, filesList, columns };
}
