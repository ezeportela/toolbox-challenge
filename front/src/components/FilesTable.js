import React from "react";

import Table from "react-bootstrap/Table";
import { useFilesTable } from "../hooks/FilesTable";
import SelectList from "./SelectList";

const App = () => {
  const { files, filesList, columns } = useFilesTable();

  return (
    <>
      {filesList.length > 0 && (
        <SelectList
          className="mb-4"
          items={filesList.map((value) => ({ name: value, value }))}
          placeholder="Select a file"
        />
      )}
      {files.length > 0 && (
        <Table responsive className="table-striped table-bordered files-table">
          <thead>
            <tr>
              {columns.map((col) => (
                <th key={col}>{col}</th>
              ))}
            </tr>
          </thead>
          <tbody>
            {files.map((file) =>
              file.lines.map((line) => (
                <tr key={line.hex}>
                  <td>{file.file}</td>
                  <td>{line.text}</td>
                  <td>{line.number}</td>
                  <td>{line.hex}</td>
                </tr>
              ))
            )}
          </tbody>
        </Table>
      )}
    </>
  );
};

export default App;
