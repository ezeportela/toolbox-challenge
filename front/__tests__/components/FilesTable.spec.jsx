import React from "react";
import FilesTable from "../../src/components/FilesTable";
import { Provider, useSelector } from "react-redux";
import configureStore from "redux-mock-store"; // If you use redux-mock-store to create a mock store
import renderer from "react-test-renderer";
import nock from "nock";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn().mockImplementation((state) => state),
}));

describe("FilesTable (Snapshot)", () => {
  it("Should render component", () => {
    nock("http://localhost:8080")
      .get("/files/data")
      .reply(200, [
        {
          file: "file.csv",
          lines: [
            {
              text: "jOVwNAfWqtHYRNlT",
              number: 16,
              hex: "202cc29bb691e6433e3684cf7daa9f4f",
            },
          ],
        },
      ]);

    const mockStore = configureStore([]);
    const initialState = {
      loading: false,
      files: [
        {
          file: "file.csv",
          lines: [
            {
              text: "jOVwNAfWqtHYRNlT",
              number: 16,
              hex: "202cc29bb691e6433e3684cf7daa9f4f",
            },
          ],
        },
      ],
    };
    useSelector.mockReturnValue(initialState);

    const store = mockStore(initialState);

    const component = renderer
      .create(
        <Provider store={store}>
          <FilesTable />
        </Provider>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
