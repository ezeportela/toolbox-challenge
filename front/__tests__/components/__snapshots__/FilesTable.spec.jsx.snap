// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesTable (Snapshot) Should render component 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table-striped table-bordered files-table table"
  >
    <thead>
      <tr>
        <th>
          File Name
        </th>
        <th>
          Text
        </th>
        <th>
          Number
        </th>
        <th>
          Hex
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          file.csv
        </td>
        <td>
          jOVwNAfWqtHYRNlT
        </td>
        <td>
          16
        </td>
        <td>
          202cc29bb691e6433e3684cf7daa9f4f
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
